### PR TITLE
icu4c 74.2

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Promising initial run.

Linux:
* `gspell` can't seem to find python3 (via glib I think).
* `manticoresearch` sha256 error.
* `planck` fails because it looks like `webkitgtk` needs a revision bump.
* `qt` (Chromium) is incompatible with ICU 74 due to the addition of five new line break types with Unicode 15.1. There'll probably be a patch soon, if not already.
* `simdutf` timed out for some reason - might need a lighter test or maybe it's signalling a genuine problem.
* `tepl` also can't find python3 (via glib too I think)
* `vte3` also can't find python3 - I reckon there's something wrong with glib and we'll need to fix that.
* `zobra` maybe segfaulting

macOS:
* `manticoresearch` sha256 error.
* `texlive` - Perl mismatch somewhere: `Parser.c: loadable library and perl binaries are mismatched (got first handshake key 0x10380080, needed 0x10200080)`

Any non-disabled PHP versioned formula also needs revision bumping here.

(Haven't checked dependent tests yet as they're still running)

**Comment by Bo98** [comment]
It'll probably just be a case of we'll do it if this PR ships before that date, and not if it doesn't. PHP's light enough to only take 10-20 minutes to build.

Focus first is fixing the other things though.

**Comment by iMichka** [comment]
Fix for `gspell`: #154128

**Comment by Bo98** [comment]
Looks like none of the PHP formulae are being revision bumped here.

**Comment by iMichka** [comment]
Linux failures:
```
2023-11-26T22:49:55.1656031Z Warning: 2 failed steps ignored!
2023-11-26T22:49:55.1658020Z brew test --verbose simdutf
2023-11-26T22:49:55.1658927Z brew install --verbose --build-bottle zorba
2023-11-26T22:49:55.1666147Z Error: 3 failed steps!
2023-11-26T22:49:55.1666866Z brew test --verbose yaz
2023-11-26T22:49:55.1667427Z brew install --verbose --build-bottle planck
2023-11-26T22:49:55.1668176Z brew install --verbose --build-bottle qt
```

**Comment by cho-m** [comment]
Ignoring missing revision bumps and failures I already fixed in referenced PRs, the main failures are:
- [ ] `qt` - another `libxml2` issue. Sadly 2.12 introduced a fair amount of breakage. Probably should try to fix outside this PR either in #155921 or a separate PR.
- [ ] `texlive` - still the Perl mismatch issue when building `HTML::TreeBuilder` (with mismatch coming from previous resource `HTML::Parser` which is dependency?). I see we have a separate PR to work on this #155568.

---

Outside those, the rest look like flaky dependent tests or other unrelated issues:
- [ ] `solr@8.11` (all macOS - skipped on Linux) - timing? Looks like we already added a timeout recently. Hard to tell exact issue as we run in background so logs get sent to separate file. On side note, it looks like retry test doesn't work as the previous test's process is still active.
- [ ] `hadoop` (14-x86_64) - timing?
- [ ] `localtunnel` (14-arm64, 13-arm64)
- [ ] `libosinfo` (13-x86_64)
- [ ] `stellar-core` (13-x86_64)
- [ ] `fastqc` (14-arm64) - maybe bad JDK. Not sure on state of OpenJDK for Sonoma given other reported issues.

**Comment by cho-m** [comment]
Added some extra revision bumps.

Dealing with patches for `qt` in separate PR #156065.

Will need another rebase before running.

**Comment by cho-m** [comment]
Linux failed to build `webkitgtk`, `planck` (due to `webkitgtk`) , and `swift`:
- [ ] `webkitgtk` - another `libxml2` issue - https://github.com/WebKit/WebKit/commit/1bad176b2496579d760852c80cff3ad9fb7c3a4b
    ```
    2023-12-03T00:43:41.6692807Z /tmp/webkitgtk-20231202-188180-corvlx/webkitgtk-2.40.5/Source/WebCore/xml/XSLStyleSheetLibxslt.cpp:143:106: error: invalid conversion from ‘void (*)(void*, xmlError*)’ {aka ‘void (*)(void*, _xmlError*)’} to ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void*, const _xmlError*)’} [-fpermissive]
    2023-12-03T00:43:41.6695985Z   143 |     XMLDocumentParserScope scope(cachedResourceLoader(), XSLTProcessor::genericErrorFunc, XSLTProcessor::parseErrorFunc, console);
    2023-12-03T00:43:41.6697143Z       |                                                                                           ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
    2023-12-03T00:43:41.6697780Z       |                                                                                                          |
    2023-12-03T00:43:41.6698650Z       |                                                                                                          void (*)(void*, xmlError*) {aka void (*)(void*, _xmlError*)}
    ```
- [ ] `swift` - it's accidentally picking up `python@3.12` from a dependency and failing with `python-setuptools`. We should use an argument to make sure the Python matches dependency we set:
    ```
    2023-12-03T17:53:20.9369231Z FAILED: lib/libc++.a 
    2023-12-03T17:53:20.9392590Z : && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.27.9/bin/cmake -E rm -f lib/libc++.a && /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/bin/llvm-ar Dqc lib/libc++.a  libcxx/src/CMakeFiles/cxx_static.dir/algorithm.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/any.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/assert.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/atomic.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/barrier.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/bind.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/charconv.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/chrono.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/condition_variable.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/condition_variable_destructor.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/exception.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/functional.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/future.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/hash.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/legacy_pointer_safety.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/memory.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/mutex.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/mutex_destructor.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/new.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/optional.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/random_shuffle.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/ryu/d2fixed.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/ryu/d2s.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/ryu/f2s.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/shared_mutex.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/stdexcept.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/string.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/system_error.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/thread.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/typeinfo.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/utility.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/valarray.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/variant.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/vector.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/debug.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/legacy_debug_handler.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/random.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/ios.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/ios.instantiations.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/iostream.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/locale.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/regex.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/strstream.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/format.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/filesystem/operations.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/filesystem/directory_iterator.cpp.o libcxx/src/CMakeFiles/cxx_static.dir/filesystem/int128_builtins.cpp.o && /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/bin/llvm-ranlib -D lib/libc++.a && cd /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/tools/clang/runtime/compiler-rt-bins/lib/fuzzer/libcxx_fuzzer_x86_64/libcxx/src && /home/linuxbrew/.linuxbrew/bin/python3.12 /tmp/swift-20231203-1836061-etto1i/llvm-project/libcxx/utils/merge_archives.py -o /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/tools/clang/runtime/compiler-rt-bins/lib/fuzzer/libcxx_fuzzer_x86_64/lib/libc++.a --ar /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/bin/llvm-ar /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/tools/clang/runtime/compiler-rt-bins/lib/fuzzer/libcxx_fuzzer_x86_64/lib/libc++.a /tmp/swift-20231203-1836061-etto1i/build/llvm-linux-x86_64/tools/clang/runtime/compiler-rt-bins/lib/fuzzer/libcxx_fuzzer_x86_64/lib/libc++abi.a
    2023-12-03T17:53:20.9392949Z Traceback (most recent call last):
    2023-12-03T17:53:20.9393908Z   File "/tmp/swift-20231203-1836061-etto1i/llvm-project/libcxx/utils/merge_archives.py", line 11, in <module>
    2023-12-03T17:53:20.9394097Z     import distutils.spawn
    2023-12-03T17:53:20.9394458Z ModuleNotFoundError: No module named 'distutils'
    ```

**Comment by cho-m** [comment]
> Why not use [`--disable-renaming`](https://unicode-org.github.io/icu/userguide/icu4c/build.html#icu-as-a-system-level-library) here?

Looking at Fedora/RedHat's attempt (https://bugzilla.redhat.com/show_bug.cgi?id=856594), it seems to have gone poorly with SONAMEs still updated and patching to pass down flags to dependents with finally reverting everything back.

Looking at Ubuntu/Debian discussion (https://bugs.launchpad.net/ubuntu/+source/icu/+bug/675946), it seems like they thought it would lead to greater instability and need of some extra hacks due to SONAME issue.

The main advantage seems to be for binary compatibility but I don't think we can guarantee all requirements in Homebrew, ref https://unicode-org.github.io/icu/userguide/icu/design#icu-binary-compatibility:
> Here are the requirements for enabling binary compatibility for ICU4C:
> 1. Applications must use only APIs that are marked as stable.
> 2. Applications must use only plain C APIs, never C++.
> 3. ICU must be built with function renaming disabled.
> 4. Applications must be built using an ICU that was configured for binary compatibility.
> 5. Use ICU version 3.0 or later.
> 6. Provide both “common” and “i18n” libraries, or build a combined library.

**Comment by p-linnane** [comment]
The LICENSE file in the 74.2 tarball is an alias instead of an actual file, leading to `install: ./../LICENSE: No such file or directory`

**Comment by Bo98** [comment]
Yeah 74.2 tarball seems busted. Looks like there may be a re-release soon: https://github.com/unicode-org/icu/pull/2749#issuecomment-1858570143

**Comment by srl295** [comment]
> The LICENSE file in the 74.2 tarball is an alias instead of an actual file, leading to `install: ./../LICENSE: No such file or directory`

an alias to nothing. Not sure what the status is here.

**Comment by Bo98** [comment]
74.2 is still marked as a pre-release on GitHub but there is now a "icu4c-74_2-src-FIXED.tgz" we can use.

**Comment by srl295** [comment]
Let me know if that works. Still getting gpg and SHA updated. I updated
those. FIXED will probably disappear.

File: Formula/i/icu4c.rb
Comment (srl295) [inline Formula/i/icu4c.rb:4]: ! please don't use this. This file will be deleted.

**Review by srl295 — COMMENTED** [review]
please don't merge with the `-FIXED` file — testing is fine.

**Comment by cho-m** [comment]
> `13_x86` looks either stuck or very slow, now finished to build after 2 days and 10H while other x86 took a little more than one day and one day and 13H.

Probably got stuck given how long it was trying to build `mysql@8.0`:
```
Mon, 25 Dec 2023 06:27:14 GMT ==> brew fetch --retry mysql@8.0 --build-bottle --force
Mon, 25 Dec 2023 06:27:23 GMT ==> brew install --verbose --build-bottle mysql@8.0
Thu, 28 Dec 2023 00:55:39 GMT Error: The operation was canceled.
```

---

Looks like dep testing was broken in this run. Was it due to disabling bottle cache?

---

For Linux:
- May need to wait for `swift` fix for reproducible Python selection (right now it picks newest Python installed). I closed my PR attempt and am leaving it to #157113. (Alternatively can just fix build with `python@3.12` if version bump is delayed)
- `samba` broke due to logic change when syntax-fixing `MacOS.version`.

File: Formula/i/icu4c.rb
Comment (srl295) [reply Formula/i/icu4c.rb:5]: @Porkepix yea so the FIXED one is not merged back into the main one yet (the hashes need to be recalculated). FYI @markusicu

Comment (srl295) [reply Formula/i/icu4c.rb:5]: Sorry! Then I have no opinion other than that it looks wrong.

Comment (p-linnane) [reply Formula/i/icu4c.rb:5]: Yes, it's incorrect. It possibly got lost during a rebase.

**Comment by tbussmann** [comment]
If the `postgresql` formulas are rebuilt with an updated version of ICU, and a user has an existing database that uses a collation based on the old ICU version, the database may break and require manual intervention. In this case, PostgreSQL will issue a warning at runtime, see [here](https://www.postgresql.org/docs/current/sql-altercollation.html#SQL-ALTERCOLLATION-NOTES).

I'm not sure how this could be dealt with properly, other than to say so in the caveats section of this formula.

**Comment by srl295** [comment]
> If the `postgresql` formulas are rebuilt with an updated version of ICU, and a user has an existing database that uses a collation based on the old ICU version, the database may break and require manual intervention. In this case, PostgreSQL will issue a warning at runtime, see [here](https://www.postgresql.org/docs/current/sql-altercollation.html#SQL-ALTERCOLLATION-NOTES).
> 
> I'm not sure how this could be dealt with properly, other than to say so in the caveats section of this formula.

so i spent a long time working on trying to avoid this kind of breakage under one API, https://unicode-org.atlassian.net/browse/ICU-6631 has the details. but the short is that it's very challenging to maintain this kind of compatibility.

is there a way postgres could depend on a specific ICU version, and refuse to allow upgrade without manual intervention?  sort of how some .deb upgrades can be 'kept back' (locked)?

**Comment by tbussmann** [comment]
There has been some discussion and work to support loading multiple versions of ICU at runtime in PostgreSQL about a year ago. See [commit fest](https://commitfest.postgresql.org/46/3956/) and [discussion](https://www.postgresql.org/message-id/flat/381977b1-0898-cb6f-a427-3b5d873e81bd@enterprisedb.com) (further down that thread).

> is there a way postgres could depend on a specific ICU version, and refuse to allow upgrade without manual intervention? sort of how some .deb upgrades can be 'kept back' (locked)?

That would be very nice and should improve situations with pinned dependencies in general. My limited knowledge of homebrew-internals does not allow to judge if such a thing exists or would be feasible.

An option could be to change `icu4c` into a versioned formula and (permanently) depend each version of `postgresql` on one version of `icu4c`.

**Comment by Bo98** [comment]
It's a trade-off a bit. PostgreSQL lifetimes are 5 years and we don't really want to be supporting old ICU versions for 5 years given there is no long-term ICU support or anything like that and we simply don't have the resources to take on the job of tracking and backporting necessary patches (e.g. https://github.com/unicode-org/icu/commit/ca631eaa2691078c7a8848409f046c407cbd623a). And even then, when you decide to update to a new major PostgreSQL with a newer ICU, I'm not entirely sure if `pg_upgrade` even would handle the changes for you anyway.

We would have "broken" PostgreSQL multiple times in the past by using the latest ICU and nobody to my knowledge has ever complained directly to us for doing so. It's probably more serious if we were dealing with automated updates on a production server but you likely wouldn't be using an evergreen package manager for that (you'll have more serious issues there in terms of breaking changes) so its usage in Homebrew is more developer focussed. If this has become unacceptable for anyone, the best approach I reckon is for interested parties to make and maintain (or hire someone to do) a custom build, e.g. via `brew extract`, and to choose the ICU version deemed most appropriate for their particular data and provide migration when the time comes to upgrade.

**Comment by srl295** [comment]
Multiple ICUs (runtime) are happily installed in the same or different directories. In fact multiple ICUs coexist within a single address space for this exact reason.

**Comment by tbussmann** [comment]
I completely agree that this is a trade-off, and can understand your reasoning for not maintaining ICU versions. Still, I hoped there could be a way to notify users of a potentially breaking change. 

I thought about a caveat in the `icu4c` formula that would only be displayed if one of the postgresql formulas is installed and ICU gets a major update. But I failed both in detecting whether the formula was being updated and in formulating an appropriate message to the user. 
<details>
<summary>This is as far as I have got:</summary>

```ruby
  def caveats
    return unless 
      (Formula["postgresql@10"].any_version_installed? || Formula["postgresql@11"].any_version_installed? || Formula["postgresql@12"].any_version_installed? ||
      Formula["postgresql@13"].any_version_installed? || Formula["postgresql@14"].any_version_installed? || Formula["postgresql@15"].any_version_installed? ||
      Formula["postgresql@16"].any_version_installed?) &&
      Formula["icu4c"].any_version_installed? && ! Formula["icu4c"].latest_version_installed?
    <<~EOS
      ICU updated to #{version.major} while you have installed PostgreSQL: 
      In case you have used an ICU collation which differs from the new one,
      you may need to reindex your database. Watch out for collation mismatch
      warnings in Postgres. 
      See https://www.postgresql.org/docs/current/sql-altercollation.html#SQL-ALTERCOLLATION-NOTES
    EOS
  end
```

</details>

Finally, I am beginning to realise that this is probably not the way to go, and that in the long run these situations need better tooling, in `pg_upgrade` and beyond.

File: Formula/i/icu4c.rb
Comment (srl295) [reply Formula/i/icu4c.rb:4]:
```suggestion
  url "https://github.com/unicode-org/icu/releases/download/release-74-2/icu4c-74_2-src.tgz"
```

**Comment by srl295** [comment]
⚠️   Hi @p-linnane and others - posting this here because it's important - as I noted above, the source filename has gone back to it's usual form.  See my correction here https://github.com/Homebrew/homebrew-core/pull/153108/files#r1486341903 (I did not check the hashes yet, but they are availble from the same place).

The 74.2 release is (finally!) out of pre-release on GitHub so it seems to be done now.

**Comment by p-linnane** [comment]
Thanks for following up @srl295. I will push that change on our next run. Our CI is bogged down right now with new major versions of Go & Rust, so once those are merged I will circle back here.

**Comment by srl295** [comment]
> Thanks for following up @srl295. I will push that change on our next run. 

Thanks for what you do.

**Comment by p-linnane** [comment]
Kicking off a new run with the corrected tarball. Will post a summary once it's complete.

**Comment by p-linnane** [comment]
This will need some more work after the run completes, so I'd rather get the new `protobuf` PR merged while this runs.

**Comment by SMillerDev** [comment]
Manticoresearch has this error both here and on my local machine without the changes
```console
/tmp/manticoresearch-20240220-49049-oeo4i3/manticoresearch-6.2.12/src/indexing_sources/source_mysql.cpp:47:20: error: use of undeclared identifier 'mysql_ssl_set'; di
d you mean 'mysql_close'?
        static decltype (&mysql_ssl_set) sph_mysql_ssl_set = nullptr;
                          ^~~~~~~~~~~~~
                          mysql_close
/opt/homebrew/opt/mysql-client/include/mysql/mysql.h:797:14: note: 'mysql_close' declared here
void STDCALL mysql_close(MYSQL *sock);
             ^
/tmp/manticoresearch-20240220-49049-oeo4i3/manticoresearch-6.2.12/src/indexing_sources/source_mysql.cpp:218:40: error: too many arguments to function call, expected 1
, have 6
                sph_mysql_ssl_set ( &m_tMysqlDriver, m_sSslKey.cstr(), m_sSslCert.cstr(), m_sSslCA.cstr(), NULL, NULL );
                ~~~~~~~~~~~~~~~~~                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 30%] Linking CXX static library libsource_pgsql.a
```

**Comment by SMillerDev** [comment]
Zorba failure on Linux
```console
  /tmp/zorba-20240217-2363463-adrtuk/zorba-3.1/src/store/naive/loader_fast.cpp:219:26: error: invalid conversion from ‘void (*)(void*, xmlErrorPtr)’ {aka ‘void (*)(void*, _xmlError*)’} to ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void*, const _xmlError*)’} [-fpermissive]
    219 |   theSaxHandler.serror = &XmlLoader::error;
        |                          ^~~~~~~~~~~~~~~~~
        |                          |
        |                          void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
  /tmp/zorba-20240217-2363463-adrtuk/zorba-3.1/src/store/naive/loader_dtd.cpp: In constructor ‘zorba::simplestore::FragmentXmlLoader::FragmentXmlLoader(zorba::store::ItemFactory*, zorba::XQueryDiagnostics*, const zorba::store::LoadProperties&, bool)’:
  /tmp/zorba-20240217-2363463-adrtuk/zorba-3.1/src/store/naive/loader_dtd.cpp:110:26: error: invalid conversion from ‘void (*)(void*, xmlErrorPtr)’ {aka ‘void (*)(void*, _xmlError*)’} to ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void*, const _xmlError*)’} [-fpermissive]
    110 |   theSaxHandler.serror = &XmlLoader::error;
        |                          ^~~~~~~~~~~~~~~~~
        |                          |
        |                          void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
  /tmp/zorba-20240217-2363463-adrtuk/zorba-3.1/src/store/naive/loader_dtd.cpp: In constructor ‘zorba::simplestore::DtdXmlLoader::DtdXmlLoader(zorba::store::ItemFactory*, zorba::XQueryDiagnostics*, const zorba::store::LoadProperties&, bool)’:
  /tmp/zorba-20240217-2363463-adrtuk/zorba-3.1/src/store/naive/loader_dtd.cpp:603:26: error: invalid conversion from ‘void (*)(void*, xmlErrorPtr)’ {aka ‘void (*)(void*, _xmlError*)’} to ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void*, const _xmlError*)’} [-fpermissive]
    603 |   theSaxHandler.serror = &XmlLoader::error;
        |                          ^~~~~~~~~~~~~~~~~
        |                          |
        |                          void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
  make[2]: *** [src/CMakeFiles/zorba_simplestore.dir/build.make:10005: src/CMakeFiles/zorba_simplestore.dir/store/naive/loader_fast.cpp.o] Error 1
```

**Comment by cho-m** [comment]
`zorba` isn't bottled on Linux so not a blocker.

`swift` is still biggest blocker. #157113 may be ready soon.

`chakra` is failing on Xcode 15 so causing failure on Ventura runner which bottled it with older CLT. Might try with `-ld_classic` to see if it helps work around failure.

**Comment by p-linnane** [comment]
Force pushed to get us caught up with `master`. I'd like to wait for the [`qt` PR](https://github.com/Homebrew/homebrew-core/pull/163443) to finish, then I'll rebase this and hopefully it should be the last run. Nothing left was critical enough where it can't be handled in follow ups.

**Review by SMillerDev — APPROVED** [review]

**Comment by cho-m** [comment]
For CI, did we ever fix the situations that https://github.com/Homebrew/homebrew-core/labels/no%20long%20build%20conflict didn't properly work? If not, may need to keep an eye out or will need to manually drop commits later (not sure how it works with new merge workflow).

Do we need https://github.com/Homebrew/homebrew-core/labels/large-bottle-upload ?

**Comment by p-linnane** [comment]
> For CI, did we ever fix the situations that [ no long build conflict](https://github.com/Homebrew/homebrew-core/issues?q=is%3Aopen+label%3A%22no+long+build+conflict%22+sort%3Aupdated-desc) Do not allow merging other pull requests when files conflict with this one didn't properly work? If not, may need to keep an eye out or will need to manually drop commits later (not sure how it works with new merge workflow).

Yes, we did. Publishing a conflicting PR now fails before it uploads it's bottles with an appropriate message.
> Do we need [ large-bottle-upload](https://github.com/Homebrew/homebrew-core/issues?q=is%3Aopen+label%3Alarge-bottle-upload+sort%3Aupdated-desc) Pull request requires a large runner to upload bottles ?

Most likely. Last time we also had to do some branch protection stuff to avoid timing out the pre-merge checks, but I will defer to @Bo98 on that.

**Comment by Bo98** [comment]
> Last time we also had to do some branch protection stuff to avoid timing out the pre-merge checks, but I will defer to @Bo98 on that.

That happens post-upload so I can handle that at the time if it's an issue.

**Comment by p-linnane** [comment]
Final run results

Linux build:
```
Warning: 2 failed steps ignored!
brew test --verbose simdutf
brew install --verbose --build-bottle zorba
Error: 1 failed step!
brew install --verbose --build-bottle swift
```
Talked with @Bo98 and `swift` is most likely having issues due to all the other stuff installed. Will do a bottle dispatch after merging this.

macOS build:
```
Warning: 1 failed step ignored!
brew linkage --cached --test --strict znc
```
Likely opportunistic linkage. I recently worked on this formula and it doesn't require `boost` to build and test successfully.

Linux deps:
```
Warning: 1 failed step ignored!
brew linkage --cached --test --strict profanity
Error: 34 failed steps!
brew linkage --test percona-xtrabackup
brew test --retry --verbose percona-xtrabackup
brew fetch --retry create-api python@3.11 swift
brew fetch --retry grpc-swift python@3.11 swift swift-protobuf
brew fetch --retry mint python@3.11 swift
brew fetch --retry mockolo python@3.11 swift
brew fetch --retry publish swift
brew fetch --retry sourcery swift
brew fetch --retry swift-format swift
brew fetch --retry swift-outdated swift
brew fetch --retry swift-protobuf swift
brew fetch --retry swift-sh swift
brew fetch --retry swiftdraw swift
brew fetch --retry swiftformat swift
brew fetch --retry swiftlint swift
brew fetch --retry vapor swift
brew fetch --retry xcbeautify swift
brew fetch --retry xclogparser swift
brew linkage --test libspelling
brew test --retry --verbose libspelling
brew test --retry --verbose openliberty-jakartaee8
brew test --retry --verbose openliberty-jakartaee9
brew test --retry --verbose openliberty-microprofile4
brew test --retry --verbose openliberty-webprofile8
brew test --retry --verbose openliberty-webprofile9
brew test --retry --verbose cgal
brew linkage --test freeling
brew test --retry --verbose freeling
brew linkage --test libphonenumber
brew test --retry --verbose libphonenumber
brew linkage --test ncmpcpp
brew test --retry --verbose ncmpcpp
brew linkage --test mapnik
brew linkage --test qt-percona-server
```
All the `swift` related stuff is failiing because `swift` didn't build. Will follow up on these and rev bump the remaining things that need it in follow up PRs.

macOS deps:
```
Warning: 1 failed step ignored!
brew linkage --cached --test --strict mapserver
Error: 14 failed steps!
brew linkage --test percona-xtrabackup
brew test --retry --verbose percona-xtrabackup
brew linkage --test libspelling
brew test --retry --verbose libspelling
brew linkage --test apngasm
brew test --retry --verbose apngasm
brew linkage --test freeling
brew test --retry --verbose freeling
brew linkage --test libphonenumber
brew test --retry --verbose libphonenumber
brew linkage --test ncmpcpp
brew test --retry --verbose ncmpcpp
brew linkage --test mapnik
brew test --retry --verbose iconsur
```
`iconsur` is already being worked on in https://github.com/Homebrew/homebrew-core/pull/163904. Will rev bump and triage the remaining stuff in follow up PRs.

**Comment by p-linnane** [comment]
Thanks to everyone who contributed to the effort to get this version bump merged!

**Pull request merged**
